src/main/java/com/epam/test_generator/services/PasswordService.java

```diff
@@ -64,18 +63,14 @@ public String createConfirmUrl(HttpServletRequest request, Token token) {
      */
     public void passwordReset(PasswordResetDTO passwordResetDTO) {
         String token = passwordResetDTO.getToken();
-        Token resetToken = tokenDAO.findByTokenUuid(token);
```

Comment:
What was the motivation of replacing `TokenDAO` with `TokenService`?

Reply:
@pomkine, `PasswordService` uses the `TokenDAO` methods, which are available in `TokenService`. I think if we have `TokenService`, an other classes should use it instead of `TokenDAO`.

Reply:
The bug is about cyclic dependencies. It happens when you are using services inside each other. Instead of moving methods between services you can move these logic into `Token` class.
Also why we delete a token from the db if it is expired?